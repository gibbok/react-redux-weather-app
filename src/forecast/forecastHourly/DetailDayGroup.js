// @flow
import * as React from 'react'
import moment from 'moment'
import DetailHour from './DetailHour'
import { withStyles, Typography, Divider } from 'material-ui'

/* eslint-disable no-undef */
type PropsType = {
  +timespan:number,
  +hours: Array<{
    +humidityUnit: '%',
    +humidityValue: number,
    +momentId: string,
    +temp: number,
    +timespan: number,
    +weatherDescription: string,
    +weatherIconCode: number,
    +windDegree: number,
    +windSpeed: number
  }>,
  +classes: Object
}
/* eslint-enable no-undef */

const styles = theme => {
  const spaceUnit = theme.spacing.unit
  return ({
    root: {
    },
    title: {
      marginBottom: spaceUnit * 1
    },
    hours: {
      display: 'flex',
      marginTop: spaceUnit * 1.5,
      marginBottom: spaceUnit * 3
      // borderTop: '1px solid rgba(0,0,0,0.5)',
      // borderBottom: '1px solid rgba(0,0,0,0.5)'
    }
  })
}

const DetailDayGroup:React.StatelessFunctionalComponent<PropsType> = ({ timespan, hours, classes }:PropsType):React.Element<any> => {
  const dayFormat = moment.unix(timespan).format('ddd, MMM D')
  const hoursDetails = hours.map((day, index) =>
    <DetailHour
      key={index}
      {...hours[index]}
    />
  )
  return (
    <div className={classes.root}>
      <div className={classes.title}>
        <Typography type='title'>
          {dayFormat}
        </Typography>
      </div>
      <Divider />
      <div className={classes.hours}>
        {hoursDetails}
      </div>
    </div>
  )
}

export default withStyles(styles)(DetailDayGroup)
