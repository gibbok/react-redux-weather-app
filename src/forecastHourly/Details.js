import React from 'react'
import Detail from './Detail'

const Details = ({ forecastHourly }) => (
  <div>
    <ul>
      {forecastHourly.map((day, index) =>
        <Detail
          key={index}
          {...forecastHourly[index]}
      />
    )}
    </ul>

  </div>
)

export default Details
