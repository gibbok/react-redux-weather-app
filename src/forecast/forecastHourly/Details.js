import React from 'react'
// import Detail from './Detail'
import DetailDayGroup from './DetailDayGroup'

/*
 {
  date1: [],
  date2: []
 }
*/

const Details = ({ forecastHourly }) => {
  let result = forecastHourly.reduce((acc, item) => {
    (acc[item.momentId] || (acc[item.momentId] = [])).push(item)
    return acc
  }, {})
  console.log(result)
  return (
    <div>
      {forecastHourly.map((day, index) =>
        <DetailDayGroup
          key={index}
          day={day}
      />
    )}
    </div>
  )
}

// const Details = ({ forecastHourly }) => (
//   <div>
//     <ul>
//       {forecastHourly.map((day, index) =>
//         <Detail
//           key={index}
//           {...forecastHourly[index]}
//       />
//     )}
//     </ul>
//   </div>
// )

export default Details
