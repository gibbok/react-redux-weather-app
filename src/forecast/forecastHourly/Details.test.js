// @flow
import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import Details from './Details'
import DetailDayGroup from './DetailDayGroup'

describe('<Details />', () => {
  it('should render', () => {
    const forecastHourly = [
      {
        momentId: '2017:8:12:2',
        timespan: 1505206800,
        temp: 14.73,
        weatherIconCode: 803,
        weatherDescription: 'Clouds',
        windDegree: 273.006,
        windSpeed: 1.26,
        humidityUnit: '%',
        humidityValue: 93},
      {
        momentId: '2017:8:13:1',
        timespan: 1505217600,
        temp: 16.55,
        weatherIconCode: 804,
        weatherDescription: 'Clouds',
        windDegree: 344,
        windSpeed: 1.3,
        humidityUnit: '%',
        humidityValue: 89}
    ]
    const wrapper = shallow(
      <Details
        forecastHourly={forecastHourly}
      />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should render two DetailDayGroup components (one for each day)', () => {
    const forecastHourly = [
      {
        momentId: '2017:8:12:2',
        timespan: 1505206800,
        temp: 14.73,
        weatherIconCode: 803,
        weatherDescription: 'Clouds',
        windDegree: 273.006,
        windSpeed: 1.26,
        humidityUnit: '%',
        humidityValue: 93},
      {
        momentId: '2017:8:13:1',
        timespan: 1505217600,
        temp: 16.55,
        weatherIconCode: 804,
        weatherDescription: 'Clouds',
        windDegree: 344,
        windSpeed: 1.3,
        humidityUnit: '%',
        humidityValue: 89}
    ]
    const wrapper = shallow(
      <Details
        forecastHourly={forecastHourly}
        />
      )
    expect(wrapper.find(DetailDayGroup).length).toBe(2)
  })
})
