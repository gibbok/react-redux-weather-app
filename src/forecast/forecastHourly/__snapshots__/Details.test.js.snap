// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Details /> should render 1`] = `
<div>
  <DetailDayGroup
    hours={
      Array [
        Object {
          "humidityUnit": "%",
          "humidityValue": 93,
          "momentId": "2017:8:12:2",
          "temp": 14.73,
          "timespan": 1505206800,
          "weatherDescription": "Clouds",
          "weatherIconCode": 803,
          "windDegree": 273.006,
          "windSpeed": 1.26,
        },
      ]
    }
    timespan={1505206800}
  />
  <DetailDayGroup
    hours={
      Array [
        Object {
          "humidityUnit": "%",
          "humidityValue": 89,
          "momentId": "2017:8:13:1",
          "temp": 16.55,
          "timespan": 1505217600,
          "weatherDescription": "Clouds",
          "weatherIconCode": 804,
          "windDegree": 344,
          "windSpeed": 1.3,
        },
      ]
    }
    timespan={1505217600}
  />
</div>
`;
